src/program.ts

```diff
@@ -3588,11 +3616,37 @@ export class Function extends TypedElement {
     return local;
   }
 
+  markClosedLocal(local: Local) {
+    console.log("marking local " + local.name + " as closed in function " + this.name);
+    local.envOffset = this.nextEnvOffset;
+    // TODO: the alignment logic is slightly more complicated then this iirc
```

Comment:
I'm pretty sure there's somewhere in the compiler that already does this I copied off last time, but it's not terribly important for this PR so I'm leaving it.